app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/attendeeRepository/remote/AttendeeApiImpl.kt

```diff
@@ -0,0 +1,42 @@
+package com.realityexpander.tasky.agenda_feature.data.repositories.attendeeRepository.remote
+
+import com.realityexpander.tasky.agenda_feature.common.util.EventId
+import com.realityexpander.tasky.agenda_feature.data.repositories.eventRepository.remote.eventApi.DTOs.AttendeeExistsResponseDTO
+import com.realityexpander.tasky.core.data.remote.TaskyApi
+import com.realityexpander.tasky.core.data.remote.utils.getErrorBodyMessage
+import com.realityexpander.tasky.core.util.Email
+import javax.inject.Inject
+
+class AttendeeApiImpl @Inject constructor(
+    private val taskyApi: TaskyApi
+) : IAttendeeApi {
+
+    override suspend fun getAttendee(email: Email): Result<AttendeeExistsResponseDTO> {
+        try {
+            val response = taskyApi.getAttendee(email)
+
+            if (response.isSuccessful) {
+                response.body()?.let { attendeeExistsResponseDTO ->
+                    if (attendeeExistsResponseDTO.doesUserExist) {
+                        return Result.success(attendeeExistsResponseDTO)
+                    } else {
+                        return Result.failure(Exception("User email does not exist"))
+                    }
+                }
+            }
+
+            return Result.failure(Exception(getErrorBodyMessage(response.errorBody()?.string())))
```

Comment:
Could it make sense to create a more specific exception here, like `NetworkException`? Otherwise, it'll be a harder to distinguish between different types of exceptions

Reply:
ok, i will add more catches for the exceptions.